screenpy/actions/silently.py

```diff
@@ -0,0 +1,126 @@
+    All of the attributes from the "duck" should be exposed by the Silently object.
+    """
+
+    def __getattr__(self, key: Any) -> Any:
+        try:
+            return getattr(self.duck, key)
+        except AttributeError as exc:
+            msg = (
+                f"{self.__class__.__name__}({self.duck.__class__.__name__}) "
+                f"has no attribute '{key}'"
+            )
+            raise AttributeError(msg) from exc
+
+
+class SilentlyPerformable(Performable, SilentlyMixin):
+    """Act like the Performable passed in but kink the cable when performing"""
+
+    def perform_as(self, actor: Actor) -> None:
+        with the_narrator.mic_cable_kinked():
+            self.duck.perform_as(actor)
+            if not settings.UNABRIDGED_LOGGING:
+                the_narrator.clear_backup()
+            return
+
+    def __init__(self, duck: Performable):
+        if not isinstance(duck, Performable):
+            raise NotPerformable(
+                "SilentlyPerformable only works with Performable. "
+                "Use `Silently` instead."
+            )
+        self.duck = duck
+
+
+class SilentlyAnswerable(Answerable, SilentlyMixin):
+    """Act like the Answerable passed in but kink the cable when answering"""
+
+    def answered_by(self, actor: Actor) -> Any:
+        with the_narrator.mic_cable_kinked():
+            thing = self.duck.answered_by(actor)
+            if not settings.UNABRIDGED_LOGGING:
+                the_narrator.clear_backup()
+            return thing
+
+    def __init__(self, duck: Answerable):
+        if not isinstance(duck, Answerable):
+            raise NotAnswerable(
+                "SilentlyAnswerable only works with Answerable. Use `Silently` instead."
+            )
+        self.duck = duck
+
+
+class SilentlyResolvable(Resolvable, SilentlyMixin):
+    """Act like the Resolvable passed in but kink the cable when resolving"""
+
+    def resolve(self) -> Matcher:
+        with the_narrator.mic_cable_kinked():
+            res = self.duck.resolve()
+            if not settings.UNABRIDGED_LOGGING:
+                the_narrator.clear_backup()
+            return res
+
+    def __init__(self, duck: Resolvable):
+        if not isinstance(duck, Resolvable):
+            raise NotResolvable(
+                "SilentlyResolvable only works with Resolvable. Use `Silently` instead."
+            )
+        self.duck = duck
+
+
+T_duck: TypeAlias = Union[Performable, Answerable, Resolvable]
+T_quack: TypeAlias = Union[SilentlyAnswerable, SilentlyPerformable, SilentlyResolvable]
+
+
+@overload
+def Silently(duck: Performable) -> Union[Performable, SilentlyPerformable]:
+    ...
+
+
+@overload
+def Silently(duck: Answerable) -> Union[Answerable, SilentlyAnswerable]:
+    ...
+
+
+@overload
+def Silently(duck: Resolvable) -> Union[Resolvable, SilentlyResolvable]:
+    ...
+
+
+def Silently(duck: T_duck) -> Union[T_duck, T_quack]:
+    """
```

Comment:
This makes little sense analogy wise.  We are returning a silent thing and yet the annotation says "quack" (which is very much the opposite of silent).